src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,140 @@
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        int oldElementDataSize = size;
+        ensureCapacityInternal(size + list.size());
+        for (int i = 0; i < list.size(); i++) {
+            elementData[oldElementDataSize++] = list.get(i);
+        }
+        size = size + list.size();
+    }
+
+    //////// Methods for addition
+
+    private void ensureCapacityInternal(int minCapacity) {
+        if (minCapacity - elementData.length > 0) {
+            grow(minCapacity);
+        }
+    }
 
+    private void grow(int minCapacity) {
+        int oldCapacity = elementData.length;
+        int newCapacity = oldCapacity + (oldCapacity >> 1);
+        if (newCapacity - minCapacity < 0) {
+            newCapacity = minCapacity;
+        }
+        if (newCapacity - MAX_ARRAY_SIZE > 0) {
+            throw new OutOfMemoryError();
+        }
+        elementData = Arrays.copyOf(elementData, newCapacity);
     }
 
+    private void rangeCheckForAdd(int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+    }
+
+    /////////////// GET - SET //////////////
+
     @Override
     public T get(int index) {
-        return null;
+        rangeCheck(index);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
+        rangeCheck(index);
+        elementData[index] = value;
+    }
 
+    private void rangeCheck(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
     }
 
+    /////////////////////// REMOVE //////////////////////////////////////
+
     @Override
     public T remove(int index) {
-        return null;
+        rangeCheck(index);
+        T oldValue = (T) elementData[index];
+        int numOfElementsMoved = size - index - 1;
+        if (numOfElementsMoved > 0) {
+            System.arraycopy(elementData, index + 1, elementData, index, numOfElementsMoved);
+        }
+        elementData[--size] = null;
+        return oldValue;
     }
 
     @Override
     public T remove(T element) {
+        if (element == null) {
+            for (int i = 0; i < size; i++) {
+                if (elementData[i] == null) {
+                    fastRemove(i);
+                    return element;
+                }
+            }
+        } else {
+            for (int i = 0; i < size; i++) {
+                if (element.equals(elementData[i])) {
+                    fastRemove(i);
+                    return element;
+                }
+            }
+            throw new NoSuchElementException("Element" + element + "is not present in ArrayList");
+        }
         return null;
     }
 
+    private void fastRemove(int index) {
```

Comment:
I do not think we need it)

Reply:
Yes, it was possible to make it shorter.